src/vs/workbench/contrib/terminalContrib/links/browser/terminalLinkDetectorAdapter.ts

```diff
@@ -93,9 +95,10 @@ export class TerminalLinkDetectorAdapter extends Disposable implements ILinkProv
 
 		const detectedLinks = await this._detector.detect(lines, startLine, endLine);
 		for (const link of detectedLinks) {
-			links.push(this._createTerminalLink(link, async (event) => this._onDidActivateLink.fire({ link, event })));
+			const terminalLink = this._createTerminalLink(link, async (event) => this._onDidActivateLink.fire({ link, event }));
+			links.push(terminalLink);
+			this._activeLinksStore.add(terminalLink);
```

Comment:
I couldn't repro the leak, but looking at the trace https://github.com/microsoft/vscode/issues/252302, here is my hypothesis:
 
The 'active links` that gets added into `_activeLinks` may not get disposed when the `TerminalLinkDetectorAdapter` itself gets disposed. 

What I did:
* Created `DisposableStore` to add in all these links the moment they are added to `_activeLinks`, so when adapter gets disposed, all of these links would get disposed too.

Concern:
* I wanted to avoid having both `activeLinks` and `_activeLinksStore`, but